app/dao/ExtractionResultDAO.scala

```diff
@@ -1,46 +1,84 @@
 package dao
 
 import javax.inject.Inject
-import models.{ExtractedAssetModel, ExtractedBlockModel, ExtractedInputModel, ExtractedOutputModel, ExtractedRegisterModel, ExtractedTransactionModel, ExtractionResultModel}
+import models.{ExtractedAssetModel, ExtractedBlockModel, ExtractedDataInputModel, ExtractedInputModel, ExtractedOutputModel, ExtractedRegisterModel, ExtractedTransactionModel, ExtractionInputResultModel, ExtractionOutputResultModel, ExtractionResultModel}
 import play.api.db.slick.{DatabaseConfigProvider, HasDatabaseConfigProvider}
 import slick.jdbc.JdbcProfile
 import utils.DbUtils
 
 import scala.concurrent.duration.Duration
-import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent.{Await, ExecutionContext}
 
 
-class ExtractionResultDAO @Inject() (extractedBlockDAO: ExtractedBlockDAO, transactionDAO: TransactionDAO, inputDAO: InputDAO, outputDAO: OutputDAO, assetDAO: AssetDAO, registerDAO: RegisterDAO, protected val dbConfigProvider: DatabaseConfigProvider) (implicit executionContext: ExecutionContext)
+class ExtractionResultDAO @Inject() (extractedBlockDAO: ExtractedBlockDAO, transactionDAO: TransactionDAO, dataInputDAO: DataInputDAO, inputDAO: InputDAO, outputDAO: OutputDAO, assetDAO: AssetDAO, registerDAO: RegisterDAO, protected val dbConfigProvider: DatabaseConfigProvider) (implicit executionContext: ExecutionContext)
     extends DbUtils with HasDatabaseConfigProvider[JdbcProfile] {
 
+  import profile.api._
   /**
   * store data into db as transactionally.
-   * @param extractionResult : ExtractionResultModel extracted data
+   * @param spentTrackedInputs : ExtractionInputResultModel extracted spent Tracked Inputs
+   */
+  def storeBaseOnInputs(spentTrackedInputs: Seq[ExtractionInputResultModel]): Unit = {
```

Comment:
Sounds like inputs for tracked outputs being spent passed, while in fact all the inputs in the block. Please fix descriptions and naming. 

Inputs filtering can be very inefficient, can we consider using and storing a Bloom filter maybe?

Reply:
I think according to the database queries, this is the only way to use Bloom filter, If you have another comment please share

Reply:
It seems Bloom filter is not doing doing much here, also, after checking with mightContain you need to check for false positives. Let's revert ?

Reply:
Revert on which? the filter or check false positives for Bloom filter? 
I think revert on the filter is better!